openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -522,6 +522,10 @@ def get_group_info_for_cohort(cohort, use_cached=False):
     use_cached=True to use the cached value instead of fetching from the
     database.
     """
+    if cohort.pk is None:
+        # Object not saved yet, skip caching
```

Comment:
```suggestion
        # Cohort not saved yet, so no PartitionGroup can be linked to it
```

Minor suggestion, I'd reword this comment. It's not just that we're skipping caching, it's that we are correctly returning with the information that no PartitionGroup can be linked to an unsaved Cohort.